zc_common/remote_resource/request.py

```diff
@@ -80,19 +80,30 @@ def add_items_from_data(self):
         map(lambda x: self.append(RemoteResourceWrapper(x)), self.data)
 
 
-def get_route_from_fk(resource_type, pk=None):
-    """Gets a fully qualified URL for a given resource_type, pk"""
-    routes = requests.get(zc_settings.GATEWAY_ROOT_PATH).json()
-
-    for route in routes.iterkeys():
-        if 'resource_type' in routes[route] and routes[route]['resource_type'] == resource_type:
-            if isinstance(pk, (list, set)):
-                expanded = '{}?filter[id__in]={}'.format(expand(route, {}), ','.join([str(x) for x in pk]))
-            else:
-                expanded = expand(route, {'id': pk})
-            return '{0}{1}'.format(routes[route]['domain'], expanded)
-
-    raise RouteNotFoundException('No route for resource_type: "{0}"'.format(resource_type))
+class RouteManager(object):
```

Comment:
@tnsengimana As I've already stated multiple times, there is no way to update the route list across multiple processes without a worker process and a store that the web processes can read from, and that is an extremely expensive and complex solution that is wholly unacceptable right now.

> You implementation assumes the GATEWAY_ROOT_PATH is always accessible which is false when running unit tests. However, by caching the first request, the test which hits GATEWAY_ROOT_PATH will then get its response cached and thus have unit tests running without blocking.

This doesn't make any sense. You're saying `GATEWAY_ROOT_PATH` is not accessible from tests, so the `RouteManager` won't work, so we should just cache the first request, which would require the gateway to be available. It's either available, or its not.

However, you do bring up a good point. During testing, we shouldn't make a request to the gateway, because then we'd be integrating against it. Instead, `get_route_from_fk` should do something locally in the case of `DEBUG = True`. I'll make that change and push it up, and that should fix your testing issue.